### PR TITLE
Add diff_query_schema function util

### PR DESCRIPTION
**Pull request by fbertsch**
I am so tired of BQ's truncated errors.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-2449)

### CODE REVIEW

File: sql/mozfun/utils/diff_query_schemas/stored_procedure.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
+CREATE OR REPLACE PROCEDURE
+  analysis.diff_query_schemas(
```
Comment (scholtzan) [inline sql/mozfun/utils/diff_query_schemas/stored_procedure.sql:2]:
```suggestion
  utils.diff_query_schemas(
```

File: sql/mozfun/utils/diff_query_schemas/stored_procedure.sql [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    OUT res ARRAY<STRUCT<i INT64, a_col STRING, b_col STRING>>
+  )
+BEGIN
+  DECLARE id STRING DEFAULT CAST(CAST(FLOOR(POW(10,10) * RAND()) AS INT64) AS STRING);
+
+  DECLARE table_a STRING DEFAULT "diff_queries_a_" || id;
+  DECLARE table_b STRING DEFAULT "diff_queries_b_" || id;
+
+  DECLARE table_a_id STRING DEFAULT "moz-fx-data-shared-prod.tmp." || table_a;
+  DECLARE table_b_id STRING DEFAULT "moz-fx-data-shared-prod.tmp." || table_b;
```
[the remark this answers is not in the document]
Comment (fbertsch) [reply sql/mozfun/utils/diff_query_schemas/stored_procedure.sql:14]: Yup, agreed, thanks for the assist

File: sql/mozfun/utils/diff_query_schemas/stored_procedure.sql [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  EXECUTE IMMEDIATE "SELECT ARRAY_AGG(STRUCT(COLUMN_NAME, ORDINAL_POSITION)) FROM mozdata.tmp.INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = 'tmp' AND TABLE_NAME = '" || table_a || "'" INTO query_a_schema;
+  EXECUTE IMMEDIATE "SELECT ARRAY_AGG(STRUCT(COLUMN_NAME, ORDINAL_POSITION)) FROM mozdata.tmp.INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = 'tmp' AND TABLE_NAME = '" || table_b || "'" INTO query_b_schema;
+
+  SET res = (
+    SELECT ARRAY_AGG(STRUCT(ORDINAL_POSITION AS i, a.COLUMN_NAME AS a_col, b.COLUMN_NAME AS b_col) ORDER BY ORDINAL_POSITION)
+    FROM UNNEST(query_a_schema) AS a
+    FULL OUTER JOIN (SELECT * FROM UNNEST(query_b_schema)) AS b
+      USING (ORDINAL_POSITION)
+  );
```
[the remark this answers is not in the document]
Comment (fbertsch) [reply sql/mozfun/utils/diff_query_schemas/stored_procedure.sql]: Good improvement suggestions, thank you!

File: sql/mozfun/utils/diff_query_schemas/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+is truncated, and the schemas of e.g. a UNION don't match.
+
+Use it like:
+```
+DECLARE res ARRAY<STRUCT<i INT64, differs BOOL, a_col STRING, a_data_type STRING, b_col STRING, b_data_type STRING>>;
+CALL mozfun.utils.diff_query_schemas("""SELECT * FROM a""", """SELECT * FROM b""", res);
+-- See entire schema entries, if you need context
+SELECT res;
+-- See just the elements that differ
+SELECT res WHERE differs;
```
Comment (sean-rose) [inline sql/mozfun/utils/diff_query_schemas/README.md:11]:
```suggestion
SELECT * FROM UNNEST(res) WHERE differs;
```

**Review by sean-rose — APPROVED** [review]

**Pull request merged**